plugins/services/src/js/columns/ServicesTableInstancesColumn.tsx

```diff
@@ -11,34 +11,49 @@ import StringUtil from "#SRC/js/utils/StringUtil";
 import { SortDirection } from "plugins/services/src/js/types/SortDirection";
 import ServiceTableUtil from "../utils/ServiceTableUtil";
 
-export function instancesRenderer(
-  service: Service | Pod | ServiceTree
-): React.ReactNode {
-  const instancesCount = service.getInstancesCount();
-  const runningInstances = service.getRunningInstancesCount();
-  const overview =
-    runningInstances === instancesCount
-      ? ` ${runningInstances}`
-      : ` ${runningInstances}/${instancesCount}`;
+const ServiceInstances = React.memo(
+  ({
+    instancesCount,
+    runningInstances
+  }: {
+    instancesCount: number;
+    runningInstances: number;
+  }) => {
+    const overview =
+      runningInstances === instancesCount
+        ? ` ${runningInstances}`
+        : ` ${runningInstances}/${instancesCount}`;
 
-  const content = !Number.isInteger(instancesCount)
-    ? EmptyStates.CONFIG_VALUE
-    : overview;
+    const content = !Number.isInteger(instancesCount)
+      ? EmptyStates.CONFIG_VALUE
+      : overview;
 
-  // L10NTODO: Pluralize
-  const tooltipContent = (
-    <Trans render="span">
-      {runningInstances} {StringUtil.pluralize("instance", runningInstances)}{" "}
-      running out of {instancesCount}
-    </Trans>
-  );
+    // L10NTODO: Pluralize
+    const tooltipContent = (
+      <Trans render="span">
+        {runningInstances} {StringUtil.pluralize("instance", runningInstances)}{" "}
+        running out of {instancesCount}
+      </Trans>
+    );
 
+    return (
+      <NumberCell>
+        <Tooltip content={tooltipContent}>
+          <span>{content}</span>
+        </Tooltip>
+      </NumberCell>
+    );
+  }
+);
+
+export function instancesRenderer(
```

Comment:
i don't like that we now always export the cellRenderer and a memoized component. but that's not to block these awesome changes.

Reply:
Can you explain why? From my point of view it looks like the renderer now does a transition from structs to primitives, so the "wiring up / business-logic" part and the component does the rendering. 

The reason why I chose to export the component as well is that it's easier to test that way. Testing that the structs are called in certain ways or respond in certain ways seemed pretty fragile to me, therefore I switched the tests to purely test what is being rendered.

Reply:
> Can you explain why?

🤔 i don't like the current factorization of column-modules.
they seem to aim to contain the stuff that's associated with a column but miss e.g. the header-data. i wonder whether we had less code and better comprehensibility if we inlined that stuff into the `ServiceTable`. but maybe that's just me liking big files to get a lot of context fast.

i realise that my concern has not to do with this PR directly, so please resolve.

Reply:
I introduced multiple column files to enable us to easily work on them without too much conflicts. I'm happy inlining them into one file tho. maybe something for spike pr 🤷‍♂️